prov/hook/trace/src/hook_trace.c

```diff
@@ -150,6 +150,36 @@ trace_cq(struct hook_cq *cq, const char *func, int line,
 	}
 }
 
+static inline void
+trace_cq_err(struct hook_cq *cq, const char *func, int line,
+             struct fi_cq_err_entry *entry,  uint64_t flags)
+{
+	char err_buf[80];
+
+	if (!fi_log_enabled(cq->domain->fabric->hprov, FI_LOG_TRACE, FI_LOG_CQ)) {
+		return;
+	}
+	if (entry->flags & FI_RECV) {
+		fi_log(cq->domain->fabric->hprov, FI_LOG_TRACE, FI_LOG_CQ,
+		       func, line,
+		       "ctx %p flags 0x%lx, len %zu buf %p data %lu tag %lu "
+		       "olen %zu err_code %d prov_err [%s]\n",
+		       entry->op_context, entry->flags, entry->len, entry->buf,
+		       entry->data, entry->tag, entry->olen,  entry->err,
+		       fi_cq_strerror(cq->hcq, entry->prov_errno, entry->err_data,
+		                      err_buf, 80));
```

Comment:
You want to call fi_cq_strerror before and pass err_buf into the fi_log message.